### PR TITLE
feat: suppress by policy for package and image scan + tests

### PR DESCRIPTION
**Pull request by NoaAzoulay**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally we encourage adding a scope to the prefix, which indicates the targeted framework, otherwise 'general' will be assumed.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
suppress by policy for package and image scan 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/bridgecrew/integration_features/features/suppressions_integration.py [hunk 1]
```diff
@@ -70,7 +71,12 @@ def _apply_suppressions_to_report(self, scan_report: Report) -> None:
         still_failed_checks = []
         still_passed_checks = []
         for check in scan_report.failed_checks + scan_report.passed_checks:
-            relevant_suppressions = self.suppressions.get(check.check_id)
+            check_id = check.check_id
+            if check.check_name == SCA_PACKAGE_SCAN_CHECK_NAME:
+                check_id = 'BC_VUL_2'
+                check.check_id = check_id
```
Comment (itai1357) [inline checkov/common/bridgecrew/integration_features/features/suppressions_integration.py:77]: it seems redundant no?

File: checkov/common/bridgecrew/integration_features/features/suppressions_integration.py [hunk 2]
```diff
@@ -70,7 +71,12 @@ def _apply_suppressions_to_report(self, scan_report: Report) -> None:
         still_failed_checks = []
         still_passed_checks = []
         for check in scan_report.failed_checks + scan_report.passed_checks:
-            relevant_suppressions = self.suppressions.get(check.check_id)
+            check_id = check.check_id
```
Comment (itai1357) [inline checkov/common/bridgecrew/integration_features/features/suppressions_integration.py:74]: I suggest to put the next logic under a method (e.g adjust_check_id) + adding doc why we have to do it + test this new small method for understanding what is expected original check_id

**Review by itai1357 — APPROVED** [review]

**Review by ayajbara — APPROVED** [review]

File: checkov/common/bridgecrew/integration_features/features/suppressions_integration.py [hunk 3]
```diff
@@ -70,7 +73,17 @@ def _apply_suppressions_to_report(self, scan_report: Report) -> None:
...
+            # in order to be able to suppress by policy we assign the relevant check id for package / image scan
+            # and avoiding licenses vulns
+            if scan_report.check_type == CheckType.SCA_PACKAGE and check.check_name == SCA_PACKAGE_SCAN_CHECK_NAME:
+                check_id = 'BC_VUL_2'
+                check.check_id = check_id
+            if scan_report.check_type == CheckType.SCA_IMAGE and check.check_name == SCA_PACKAGE_SCAN_CHECK_NAME:
+                check_id = 'BC_VUL_1'
+                check.check_id = check_id
+
+            relevant_suppressions = self.suppressions.get(check_id)
```
Comment (itai1357) [inline checkov/common/bridgecrew/integration_features/features/suppressions_integration.py:86]:
```suggestion
            # in order to be able to suppress by policy we assign the relevant check id for package / image scan
            # and avoiding licenses vulns
            if scan_report.check_type == CheckType.SCA_PACKAGE and check.check_name == SCA_PACKAGE_SCAN_CHECK_NAME:
                check.check_id = check_id
            if scan_report.check_type == CheckType.SCA_IMAGE and check.check_name == SCA_PACKAGE_SCAN_CHECK_NAME:
                check.check_id = check_id

            relevant_suppressions = self.suppressions.get(check.check_id)
```

Comment (NoaAzoulay) [reply checkov/common/bridgecrew/integration_features/features/suppressions_integration.py:86]: 💯

**Pull request merged**